From 7062c4cd0e110af1bbf165db42c11ca2cbb50fc1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@redhat.com>
Date: Thu, 4 Jan 2018 20:19:02 +0100
Subject: [PATCH 1/2] i386: update ssdt-misc.hex.generated
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

RH-Author: Marc-André Lureau <marcandre.lureau@redhat.com>
Message-id: <20180104201902.4364-1-marcandre.lureau@redhat.com>
Patchwork-id: 78516
O-Subject: [RHEL-7.5 qemu-kvm PATCH] i386: update ssdt-misc.hex.generated
Bugzilla: 1411490
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>

RHEL commit d7b246e19d4e81f231b3aff6c3885c325be9a9d2 "i386: expose
fw_cfg QEMU0002 in SSDT" modified ssdt-misc.dsl to export fw_cfg in
ACPI tables. However, the file isn't compiled in RHEL.

Use known good version acpica-tools-20150619-3.el7.x86_64 to build the
new compiled version. I verified with a RHEL5 guest that #1377087
isn't happening again after this update.

RHEL only: This is needed so kernel module can find the device and
load the driver. Upstream qemu uses different API to build ACPI
tables.

Signed-off-by: Marc-André Lureau <marcandre.lureau@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/ssdt-misc.hex.generated | 68 ++++++++++++++++++++++++++++++++++++++---
 1 file changed, 64 insertions(+), 4 deletions(-)

diff --git a/hw/i386/ssdt-misc.hex.generated b/hw/i386/ssdt-misc.hex.generated
index 86c5725..52ad0b7 100644
--- a/hw/i386/ssdt-misc.hex.generated
+++ b/hw/i386/ssdt-misc.hex.generated
@@ -18,12 +18,12 @@ static unsigned char ssdp_misc_aml[] = {
 0x53,
 0x44,
 0x54,
-0x62,
+0x9e,
 0x1,
 0x0,
 0x0,
 0x1,
-0x80,
+0xbf,
 0x42,
 0x58,
 0x50,
@@ -176,6 +176,66 @@ static unsigned char ssdp_misc_aml[] = {
 0x0,
 0x0,
 0x10,
+0x3b,
+0x5c,
+0x2e,
+0x5f,
+0x53,
+0x42,
+0x5f,
+0x50,
+0x43,
+0x49,
+0x30,
+0x5b,
+0x82,
+0x2e,
+0x46,
+0x57,
+0x43,
+0x46,
+0x8,
+0x5f,
+0x48,
+0x49,
+0x44,
+0xd,
+0x51,
+0x45,
+0x4d,
+0x55,
+0x30,
+0x30,
+0x30,
+0x32,
+0x0,
+0x8,
+0x5f,
+0x53,
+0x54,
+0x41,
+0xa,
+0xb,
+0x8,
+0x5f,
+0x43,
+0x52,
+0x53,
+0x11,
+0xd,
+0xa,
+0xa,
+0x47,
+0x1,
+0x10,
+0x5,
+0x10,
+0x5,
+0x1,
+0xc,
+0x79,
+0x0,
+0x10,
 0x40,
 0xc,
 0x5c,
@@ -369,8 +429,8 @@ static unsigned char ssdp_misc_aml[] = {
 0x4d,
 0x58
 };
-static unsigned char ssdt_isa_pest[] = {
-0xd0
+static unsigned short ssdt_isa_pest[] = {
+0x10c
 };
 static unsigned char acpi_s4_name[] = {
 0x88
-- 
1.8.3.1

